content/articles/2019-05-vroom-1-0-0.Rmarkdown

```diff
@@ -0,0 +1,450 @@
+function, but for many datasets vroom can read them much, much
+faster (hence the name).
+
+The main reason vroom can be faster is because character data is read from the
+file lazily; you only pay for the data you use. This lazy access is done
+automatically, so no changes to your R data manipulation code are needed.
+
+vroom also provides efficient multi-threaded writing that is multiple times
+faster on most inputs than the readr writer.
+
+Install vroom with:
+
+```{r, eval = FALSE}
+install.packages("vroom")
+```
+
+The best way to get acquainted with the package is the [getting
+started](http://vroom.r-lib.org/articles/vroom.html) vignette.
+
+## Reading delimited files
+
+Compared to readr, the first difference you may note is you use only one
+function to read the files, `vroom()`. This is because `vroom()` guesses the
+delimiter of the file automatically (based on the first few lines). This works
+well most of the time, but may fail to guess properly in some cases. The
+`delim` argument is used to specify the delimiter of the file explicitly.
+
+```{r}
+library(vroom)
+
+data <- vroom("flights.tsv")
+```
+
+The summary message after reading also differs from readr. We hope this output
+gives a more informative indication if the types of your columns are being guessed
+properly. However you can still retrieve and print the full column specification
+with `spec()`.
+
+```{r}
+spec(data)
+```
+
+The message will be disabled if you supply a column specification to `col_types` when reading.
+
+```{r}
+s <- spec(data)
+
+data <- vroom("flights.tsv", col_types = s)
+```
+
+## Reading multiple files
+
+One feature new to vroom is built-in support for reading sets of files with the
+same columns into one table. Just pass the filenames to be read directly to
+vroom.
+
+```{r}
+# Split the flights data by carrier
+purrr::iwalk(
+  split(flights, flights$carrier),
+  ~ vroom::vroom_write(.x, glue::glue("flights_{.y}.tsv"), delim = "\t")
+)
+```
+
+Then we can efficiently read all of the files into one tibble by passing the filenames
+directly to `vroom()`.
+
+```{r}
+files <- fs::dir_ls(glob = "flights_*tsv")
+files
+data <- vroom(files)
+```
+
+## Reading and writing compressed files
+
+Just like readr, vroom automatically reads and writes zip, gzip, bz2 and xz compressed
+files with the standard file extensions.
+
+```{r}
+vroom_write(flights, "flights.tsv.gz")
+
+# Check file sizes to show file is compressed
+fs::file_size(c("flights.tsv", "flights.tsv.gz"))
+
+# Read the file back in
+data <- vroom("flights.tsv.gz")
+```
+
+# Reading remote files
+
+vroom can also read files from the internet as well by passing the URL of the file to `vroom()`.
+
+```{r}
+file <- "https://raw.githubusercontent.com/r-lib/vroom/master/inst/extdata/mtcars.csv"
+data <- vroom(file)
+```
+
+It can even read gzipped files from the internet (although currently not the other compressed formats).
+
+## Reading and writing from pipe connections
```

Comment:
Personally, I don't think this is important enough to include in the announcement, and including live code in data import makes me nervous.